west.yml: Bump TF-M to allow FOTA upgrade

nandojve opened the pull request:
This pull a regression fix to allow Zephyr v3.7.1 upgrade to v4.2.0.

More details in:
  ~https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/147~
  https://github.com/zephyrproject-rtos/trusted-firmware-m/pull/152

TF-M regression introduced in:
  https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/31755

Proposed fix:
https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/42130

Final fix:
https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/42981

nandojve committed:
west.yml: Bump TF-M to fix FOTA upgrade regression

Update TF-M to include fixes restoring FOTA upgrade compatibility
between Zephyr v3.7.1 and v4.3.0. This fix allow both forward and
rollback upgrades.

Signed-off-by: BUDKE Gerson Fernando <gerson.budke@leica-geosystems.com>

tomi-font reviewed (changes requested):
Fix the commit message when updating the revision, thanks!

nandojve commented:
> Fix the commit message when updating the revision, thanks!

Fixed!

tomi-font reviewed (changes requested):
Commit message is not fixed:
> **This pull a regression pull the fixes** to allow Zephyr v3.7.1 upgrade to
v4.3.0.

Also maybe this would be worth a migration guide/release notes entry because if FOTA was broken when updating from <= v3.7 to < v4.3 then maybe now it won't work when updating from Zephyr versions 4.0-4.2 to >= 4.3?

nandojve committed:
docs: migration-guide-4.3: Add TF-M attestation issue

Add documentation describing a TF-M attestation issue that prevents
firmware-over-the-air (FOTA) upgrades between affected Zephyr releases.
Recommend a direct upgrade to Zephyr v4.3.0 to avoid attestation and
upgrade compatibility issues.

Signed-off-by: BUDKE Gerson Fernando <gerson.budke@leica-geosystems.com>

tomi-font commented on doc/releases/migration-guide-4.3.rst:
```diff
@@ -537,6 +537,17 @@ Trusted Firmware-M
+  upgraded to any later TF-M version without encountering failures.
+  This limitation affects Zephyr versions using TF-M v2.1.0 through v2.1.2 specifically,
+  Zephyr v3.7 through v4.2, preventing seamless upgrades between these releases.
+  The issue was resolved in the TF-M mainline as of October 25 and the fix is included
```
```suggestion
  The issue was resolved in mainline TF-M as of October 25 and the fix is included
```

tomi-font commented on doc/releases/migration-guide-4.3.rst:
```diff
@@ -537,6 +537,17 @@ Trusted Firmware-M
+  Zephyr v3.7 through v4.2, preventing seamless upgrades between these releases.
+  The issue was resolved in the TF-M mainline as of October 25 and the fix is included
+  in Zephyr v4.3.0. Users are advised to migrate directly from any earlier Zephyr
+  release to Zephyr v4.3.0 to ensure full TF-M attestation functionality and upgrade
```
```suggestion
  release to Zephyr v4.3.0 or later to ensure full TF-M attestation functionality and upgrade
```
